src/tablemap.ts

```diff
@@ -35,40 +54,54 @@ if (typeof WeakMap != 'undefined') {
   };
 }
 
+/**
+ * @public
+ */
 export class Rect {
-  constructor(left, top, right, bottom) {
-    this.left = left;
-    this.top = top;
-    this.right = right;
-    this.bottom = bottom;
-  }
+  constructor(
+    public left: number,
+    public top: number,
+    public right: number,
+    public bottom: number,
+  ) {}
 }
```

Comment:
In my options, the Rect should not be treated as an interface instead of a class. It's a class now just because it's a little shorter to write (`return new Rect(1, 2, 3, 4)` vs `return {left: 1, top: 2, right: 3, bottom: 4}`). We could replace this class by an interface type, and also directly return literal objects. 

By the way, `prosemirror-view` already exports a `Rect` interface. Maybe we should re-use and re-export that one?

Reply:
`Rect` exists in `prosemirror-view` but it actually isn't part of public API: https://github.com/ProseMirror/prosemirror-view/blob/b6841fa6935516a91fe96119d90b1d972992a10c/src/index.ts#L4

I did convert it to a literal instead of class though